qa/rpc-tests/wallet.py

```diff
@@ -55,9 +55,11 @@ def run_test (self):
         assert_equal(len(self.nodes[1].listunspent()), 1)
         assert_equal(len(self.nodes[2].listunspent()), 0)
 
-        # Send 21 BTC from 0 to 2 using sendtoaddress call.
-        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), 11)
+        # Send 11 BTC from 0 to 2 using sendtoaddress call.
+        addr = self.nodes[2].getnewaddress()
+        self.nodes[0].sendtoaddress(addr, 11)
         self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), 10)
+        assert(self.nodes[2].validateaddress(addr)["hdkeypath"] == "m/0'/0'/1'")
```

Comment:
Not sure about this one but I feel we could create a fresh rpc-test/hd-wallet.py or wallet-hd.py, as I expect the hd functionality will be extended further.

In the future it would be easier to extend the hd test if it was a separate file and we wouldn't need to fiddle with the already bloated wallet.py